tsconfig.json

```diff
@@ -0,0 +1,102 @@
+{
```

Comment:
Typedoc needs to read a tsconfig file. Since we will generate docs from multiple packages in the monorepo, it is better to have a base tsconfig at root, any package that uses Typescript can `extends` from this config. See the extends [here](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/663/files#diff-9d0a1d34484991c176e17efb6031749c11699e6c7e93c4569a92f7846d075966R2)

Reply:
Currently our mono-repo isn't an "all TypeScript" repo. We have a mix of JavaScript and TypeScript. Although we are in the process of converting some of those projects from JavaScript to TypeScript, I don't believe that we are saying that we want our mono-repo to only be projects that are written in TypeScript. So there is always a change to have a project that is written in a different language.

This means that project "A" could implement its versions of the `npm run build:docs` differently than project "B". But we are doing a couple of things, 1. we are defining the implementation of the `build-and-release-docs.js` at the root of the mono-repo, which is why I think you have the tsconfig/tsdoc at the root as well 2. we are specifically using the `pwa-kit-react-sdk` as the definitive package to determine the version check. That might be true because all our projects are released at the same time, but that would immediately break we change how our versions are locked. 

I'd recommend breaking out the logic that determines if the version of a project is the latest, and putting that in the root scripts folder. Then each project would have it's own way of implementing the build and deploy, at that point we'd move the tsconfig/tsdoc into the project

Reply:
@wjhsf  We had a chat about this, it seems like we still need to have a discussion around how we want to host the docs for the entire monorepo, and it might not be on github pages. For the scope of this work, it is best to set up docs for `commerce-sdk-react` and host the docs on github page for now.